### DISCUSSION TITLE
docker swarm definition files for edgex foundry

### DISCUSSION THREAD

**Discussion by girigupt** [edgexfoundry/community#106] [Q&A] [answered]
I am evaluating docker swarm  for edgex foundry deployment so we can scale across machines. Is there an example swarm configuration on top of docker-compose that can help me get started faster? Any opinion on swarm vs other container management solutions?

**Comment by bnevis-i** [comment]
We've tried it, but there are so many compatibility issues with the docker-compose running in swarm mode that it is untenable to maintain it.  The biggest problem with swarm mode is it handling (or lack thereof) volumes across swarm members.  Moreover, Docker has pretty much conceded that Kubernetes is the winner in the multi-node orchestration space.

I would suggest one of the many available kubernetes distributions to use with the example helm chart, not officially supported, at https://github.com/edgexfoundry/edgex-examples/tree/main/deployment/helm.   For example, k3s, microk8s, etc.  There are also some CSP specific distributions that are designed for IoT, such as OpenYurt, that have custom EdgeX deployments.

**Reply by girigupt** [reply, marked as answer]
Thanks  @bnevis-i , this is real helpful as we evaluate our options. Will get back if I have further queries
